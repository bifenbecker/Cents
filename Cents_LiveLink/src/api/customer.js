import httpClient from "./httpClient";

export const createCustomer = ({fullName, phoneNumber}) => {
  return httpClient({
    method: "POST",
    url: "/live-status/customer",
    data: {fullName, phoneNumber},
  });
};

export const verifyCustomer = ({phoneNumber}) => {
  return httpClient({
    method: "GET",
    url: "/live-status/customer/verify",
    params: {phoneNumber},
  });
};

export const addCustomerAddress = data => {
  return httpClient({
    method: "POST",
    url: "/live-status/customer/address/create",
    data,
  });
};

export const updateCustomerAddress = data => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/customer/address/update",
    data,
  });
};

export const updateCustomerPhoneNumber = data => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/customer/phone/update",
    data,
  });
};

export const saveDeliveryInstructions = data => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/customer/address/instructions/update",
    data,
  });
};

export const getCustomerPaymentMethods = centsCustomerId => {
  return httpClient({
    method: "GET",
    url: `/live-status/customer/payment/${centsCustomerId}/payment-methods`,
  });
};

export const addCustomerPaymentMethod = data => {
  return httpClient({
    method: "POST",
    url: "/live-status/payment-methods/create",
    data,
  });
};

export const updateCustomerNotes = (storeCustomerId, data) => {
  return httpClient({
    method: "PATCH",
    url: `/live-status/customer/notes/${storeCustomerId}/update`,
    data,
  });
};
