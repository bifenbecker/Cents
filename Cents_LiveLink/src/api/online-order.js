import httpClient from "./httpClient";

export const getAvailableNearStores = ({
  businessId,
  timeZone,
  zipCode,
  lat,
  lng,
  googlePlacesId,
}) => {
  return httpClient({
    method: "GET",
    url: "/live-status/near-stores",
    params: {businessId, timeZone, zipCode, lat, lng, googlePlacesId},
  });
};

export const fetchServicesAndModifiers = (storeId, zipCode) => {
  return httpClient({
    method: "GET",
    url: `/live-status/stores/${storeId}/deliverable-services`,
    params: {zipCode},
  });
};

export const fetchFeaturedServices = (storeId, data) => {
  return httpClient({
    method: "GET",
    url: `/live-status/stores/${storeId}/featured-services`,
    params: data,
  });
};

export const fetchCustomerLatestAddress = () => {
  return httpClient({
    method: "GET",
    url: "/live-status/customer/addresses/latest",
  });
};

export const saveAddressInfo = (placeId, data) => {
  return httpClient({
    method: "PUT",
    url: `/live-status/customer/address/${placeId}`,
    data,
  });
};

export const deleteCustomerAddress = (id) => {
  return httpClient({
    method: "DELETE",
    url: `/live-status/customer/address/${id}`,
  });
};

export const fetchCustomerInfo = (storeId) => {
  return httpClient({
    method: "GET",
    url: `/live-status/customer/${storeId}/info`,
  });
};

export const fetchAvailableDeliverySettings = (storeId) => {
  return httpClient({
    method: "GET",
    url: `/live-status/stores/${storeId}/available-delivery-settings`,
  });
};

export const fetchGeneralDeliverySettings = (storeId) => {
  return httpClient({
    method: "GET",
    url: `/live-status/stores/${storeId}/general-delivery-settings`,
  });
};

export const fetchOwnDriverDeliverySettings = (storeId, params) => {
  return httpClient({
    method: "GET",
    params,
    url: `/live-status/stores/${storeId}/own-driver-delivery-settings`,
  });
};

export const fetchOnDemandDeliverySettings = (storeId) => {
  return httpClient({
    method: "GET",
    url: `/live-status/stores/${storeId}/on-demand-delivery-settings`,
  });
};

export const submitPickupOrder = (storeId, data) => {
  return httpClient({
    method: "POST",
    url: `/live-status/stores/${storeId}/order`,
    data,
  });
};

export const fetchOwnDriverDeliveryWindows = (storeId, params) => {
  return httpClient({
    method: "GET",
    params,
    url: `/live-status/stores/${storeId}/own-delivery-windows`,
  });
};

export const fetchPreferencesChoices = (businessId) => {
  return httpClient({
    method: "GET",
    url: `/live-status/preferences-choices/${businessId}`,
  });
};

export const updateStoreCustomerPreference = (businessId, data) => {
  return httpClient({
    method: "PATCH",
    url: `/live-status/customer-store/${businessId}`,
    data,
  });
};

export const fetchPreferences = (businessId) => {
  return httpClient({
    method: "GET",
    url: `/live-status/preference-choices/business/${businessId}`,
  });
};

export const updateCustomerOptionSelection = (selectionId, data) => {
  return httpClient({
    method: "PATCH",
    url: `/live-status/preference-choices/selections/${selectionId}`,
    data,
  });
};

export const createCustomerOptionSelection = (data) => {
  return httpClient({
    method: "POST",
    url: "/live-status/preference-choices/selections",
    data,
  });
};

export const deleteCustomerOptionSelection = (selectionId) => {
  return httpClient({
    method: "DELETE",
    url: `/live-status/preference-choices/selections/${selectionId}`,
  });
};

export const fetchTurnaroundTimesForCategories = (businessId) => {
  return httpClient({
    method: "GET",
    params: {businessId},
    url: "/live-status/categories/turnaround-time",
  });
};

export const getServiceTypeAvailability = (storeId) => {
  return httpClient({
    method: "GET",
    url: `/live-status/stores/${storeId}/service-availability/get`,
  });
};

export const getBusinessByCustomUrl = (customUrl) => {
  return httpClient({
    method: "GET",
    url: `/live-status/business/custom/${customUrl}`,
  });
};

export const fetchInitialOrderData = (businessId, storeId) => {
  return httpClient({
    method: "GET",
    url: "/live-status/initial-order-data",
    params: {business: businessId, store: storeId},
  });
};

export const fetchReturnWindows = (params) => {
  return httpClient({
    method: "GET",
    url: "/live-status/return-windows",
    params,
  });
};
