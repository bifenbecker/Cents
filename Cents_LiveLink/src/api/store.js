import httpClient from "./httpClient";

export const fetchStoreTheme = storeId => {
  return httpClient({
    method: "GET",
    url: `/live-status/store-theme/${storeId}`,
  });
};
