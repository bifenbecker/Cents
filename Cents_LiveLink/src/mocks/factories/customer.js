export const VerifiedCustomer = {
  id: 1,
  firstName: "first",
  lastName: "last",
  phoneNumber: "1234567890",
  otp: "111111",
};

export const UnVerifiedCustomer = {
  id: 2,
  firstName: "first",
  lastName: "last",
  phoneNumber: "9999999999",
  otp: "222222",
};

export const ErrorCustomer = {
  id: 3,
  firstName: "first",
  lastName: "last",
  phoneNumber: "1111111111",
};
