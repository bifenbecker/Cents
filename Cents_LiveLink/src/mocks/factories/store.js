export const Store = {
  id: 1,
};
