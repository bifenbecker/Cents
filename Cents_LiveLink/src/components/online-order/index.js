import {Redirect, Route, Switch, useRouteMatch} from "react-router-dom";
import {useSelector} from "react-redux";
import {createTheme, ThemeProvider} from "@material-ui/core/styles";
import {PrivateRoute} from "hoc";
import {BusinessWithBusinessId} from "./business";
import {OnlineOrderWithBusinessId} from "./business/online-order-form";
import {getOrderInitialData} from "./redux/selectors";
import {applyTheme, getFilterClass} from "utils/theme";
import defaultTheme from "theme";
import {typography} from "components/online-order/orderTheme";
import {ScheduleMain} from "./business/schedule/ScheduleMain";

const OnlineOrder = () => {
  const {path} = useRouteMatch();

  const {
    data: {theme},
  } = useSelector(getOrderInitialData);
  const filterClass = getFilterClass(theme?.primaryColor);
  const joinedTheme = applyTheme(defaultTheme, theme);
  const muiTheme = createTheme({...joinedTheme, ...typography, filterClass});

  return (
    <ThemeProvider theme={muiTheme}>
      <Switch>
        <Route
          path={`${path}/business/:businessId`}
          exact
          component={BusinessWithBusinessId}
        />
        <PrivateRoute
          path={`${path}/business/:businessId/schedule`}
          exact
          component={ScheduleMain}
        />
        <PrivateRoute
          path={`${path}/business/:businessId/new`}
          exact
          component={OnlineOrderWithBusinessId}
        />
        <Route
          path={`${path}/business/:businessId/:storeId`}
          exact
          component={BusinessWithBusinessId}
        />
        <PrivateRoute
          path={`${path}/business/:businessId/:storeId/schedule`}
          exact
          component={ScheduleMain}
        />
        <PrivateRoute
          path={`${path}/business/:businessId/:storeId/new`}
          exact
          component={OnlineOrderWithBusinessId}
        />
        <Redirect to="/" />
      </Switch>
    </ThemeProvider>
  );
};

export default OnlineOrder;
