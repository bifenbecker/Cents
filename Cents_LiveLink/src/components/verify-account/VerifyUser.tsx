import {Dispatch, SetStateAction} from "react";
import {Flex, Text, Button, Image} from "rebass/styled-components";
import {requestOtp} from "../../api/auth";
import {IllustrationPhoneOTP} from "../../assets/images";
import {useHookstate} from "@hookstate/core";
import {onlineOrderState} from "../../state/online-order";

interface VerifyUserProps {
  phoneNumber: string;
  onConfirmation: () => void;
  setLoading: Dispatch<SetStateAction<boolean>>;
  storeId: number | null;
  setOTP: Dispatch<SetStateAction<string>>;
}

const VerifyUser = ({
  onConfirmation,
  setLoading,
  phoneNumber,
  storeId,
  setOTP,
}: VerifyUserProps) => {
  const isAuthorized = useHookstate(onlineOrderState.isAuthorized).value;

  const onSubmit = async () => {
    if (isAuthorized) {
      setLoading(true);
      try {
        const res = await requestOtp({phoneNumber, storeId, isAuthorized});
        setOTP(res.data.otpCode);
      } catch (error) {
        console.log("err: ", error);
      } finally {
        setLoading(false);
      }
    }
    onConfirmation();
  };

  const phoneLastFourDigits = phoneNumber?.slice(-4);

  return (
    <>
      <Flex {...styles.content}>
        <Image src={IllustrationPhoneOTP} />
        <Text {...styles.commonText}>Making sure it's you</Text>
        <Text {...styles.text} {...styles.commonText}>
          We will send a text with a one time password to *** *** {phoneLastFourDigits}.
        </Text>
      </Flex>
      <Flex {...styles.footer.wrapper}>
        <Button variant="primary" {...styles.footer.button} onClick={onSubmit}>
          {isAuthorized ? "SHOW" : "SEND"} VERIFICATION CODE
        </Button>
      </Flex>
    </>
  );
};

const styles = {
  content: {
    flexDirection: "column" as any,
    alignItems: ["center", "center", "center"],
    justifyContent: "center" as any,
    width: "100%",
    height: "100%",
  },
  text: {
    fontFamily: "secondary",
    sx: {
      fontWeight: "normal",
    },
  },
  footer: {
    wrapper: {
      width: "100%",
      mt: "auto",
      alignItems: "center",
      justifyContent: "center",
      pt: "18px",
    },
    button: {
      width: "100%",
      height: "48px",
      fontSize: "16px",
      sx: {
        textTransform: "uppercase",
      },
    },
  },
  commonText: {
    mb: "12px",
    mt: "12px",
    textAlign: "center" as any,
  },
};

export default VerifyUser;
