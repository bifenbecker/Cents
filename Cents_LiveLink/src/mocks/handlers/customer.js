import {rest} from "msw";
import {VerifiedCustomer} from "../factories/customer";

const customerHandlers = [
  rest.get(
    `${process.env.REACT_APP_BASE_URL}live-status/customer/verify`,
    (req, res, ctx) => {
      const phoneNumber = req.url.searchParams.get("phoneNumber");

      return res(
        phoneNumber === VerifiedCustomer.phoneNumber
          ? ctx.json({
              isVerified: true,
              firstName: VerifiedCustomer.firstName,
            })
          : ctx.json({isVerified: false})
      );
    }
  ),
];

export default customerHandlers;
