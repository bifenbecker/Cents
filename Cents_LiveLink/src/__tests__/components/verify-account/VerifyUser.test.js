import React from "react";
import {render, fireEvent, waitFor} from "@testing-library/react";

import {ErrorCustomer, VerifiedCustomer} from "../../../mocks/factories/customer";
import {Store} from "../../../mocks/factories/store";
import {onlineOrderState} from "../../../state/online-order";

import VerifyUser from "../../../components/verify-account/VerifyUser";

const renderVerifyUserScreen = props => {
  const {getByText} = render(<VerifyUser {...props} />);

  return {getByText};
};

describe("VerifyUser Component", () => {
  describe("when user is not authorized(admin)", () => {
    beforeAll(() => {
      onlineOrderState.set({isAuthorized: false});
    });

    afterAll(() => {
      onlineOrderState.set({});
    });

    it("should match snapshot", () => {
      expect(render(<VerifyUser contactLastFourDigits="1234" />)).toMatchSnapshot();
    });

    it("should have a SEND VERIFICATION CODE button", async () => {
      const onConfirmation = jest.fn();
      const {getByText} = renderVerifyUserScreen({onConfirmation});

      const button = await getByText("SEND VERIFICATION CODE");

      expect(button).toBeInTheDocument();
    });

    it("should expect onConfirmation to be called after button click", async () => {
      const onConfirmation = jest.fn();
      const setLoading = jest.fn();
      const {getByText} = renderVerifyUserScreen({onConfirmation, setLoading});

      const button = await getByText("SEND VERIFICATION CODE");

      fireEvent.click(button);

      await waitFor(() => expect(setLoading).toBeCalledTimes(0));
      expect(onConfirmation).toBeCalledTimes(1);
    });
  });

  describe("when user is authorized(admin)", () => {
    beforeAll(() => {
      onlineOrderState.set({isAuthorized: true});
    });

    afterAll(() => {
      onlineOrderState.set({});
    });

    it("should match snapshot", () => {
      expect(render(<VerifyUser contactLastFourDigits="1234" />)).toMatchSnapshot();
    });

    it("should have a SHOW VERIFICATION CODE button", async () => {
      const onConfirmation = jest.fn();
      const {getByText} = renderVerifyUserScreen({onConfirmation});

      const button = await getByText("SHOW VERIFICATION CODE");

      expect(button).toBeInTheDocument();
    });

    it("should expect OTP to be set and onConfirmation to be called after button click", async () => {
      const onConfirmation = jest.fn();
      const setLoading = jest.fn();
      const setOTP = jest.fn();
      const {getByText} = renderVerifyUserScreen({
        phoneNumber: VerifiedCustomer.phoneNumber,
        storeId: Store.id,
        setLoading,
        setOTP,
        onConfirmation,
      });

      const button = await getByText("SHOW VERIFICATION CODE");

      fireEvent.click(button);

      await waitFor(() => expect(setLoading).toBeCalledTimes(2));
      await waitFor(() => expect(setLoading).toHaveBeenNthCalledWith(1, true));
      await waitFor(() => expect(setLoading).toHaveBeenNthCalledWith(2, false));
      await waitFor(() => expect(setOTP).toHaveBeenCalledWith(VerifiedCustomer.otp));
      await waitFor(() => expect(onConfirmation).toBeCalledTimes(1));
    });

    it("should do nothing if API fails", async () => {
      const onConfirmation = jest.fn();
      const setLoading = jest.fn();
      const setOTP = jest.fn();
      const {getByText} = renderVerifyUserScreen({
        phoneNumber: ErrorCustomer.phoneNumber,
        storeId: Store.id,
        setLoading,
        setOTP,
        onConfirmation,
      });

      const button = await getByText("SHOW VERIFICATION CODE");

      fireEvent.click(button);

      await waitFor(() => expect(setLoading).toBeCalledTimes(2));
      await waitFor(() => expect(setLoading).toHaveBeenNthCalledWith(1, true));
      await waitFor(() => expect(setLoading).toHaveBeenNthCalledWith(2, false));
      await waitFor(() => expect(setOTP).toBeCalledTimes(0));
    });
  });
});
