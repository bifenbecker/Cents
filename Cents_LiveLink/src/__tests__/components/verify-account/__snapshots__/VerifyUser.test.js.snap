// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyUser Component when user is authorized(admin) should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ sc-gsnTZi doahKO hzPtvw"
        height="100%"
        width="100%"
      >
        <img
          class="sc-bczRLJ iQjTFr"
          src="Illustration_Phone-OTP.svg"
        />
        <div
          class="sc-bczRLJ bOThQT"
        >
          Making sure it's you
        </div>
        <div
          class="sc-bczRLJ fyHqCQ"
          font-family="secondary"
        >
          We will send a text with a one time password to *** *** 
          .
        </div>
      </div>
      <div
        class="sc-bczRLJ sc-gsnTZi jIxXay hzPtvw"
        width="100%"
      >
        <button
          class="sc-bczRLJ iElVFP"
          font-size="16px"
          height="48px"
          width="100%"
        >
          SHOW
           VERIFICATION CODE
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ sc-gsnTZi doahKO hzPtvw"
      height="100%"
      width="100%"
    >
      <img
        class="sc-bczRLJ iQjTFr"
        src="Illustration_Phone-OTP.svg"
      />
      <div
        class="sc-bczRLJ bOThQT"
      >
        Making sure it's you
      </div>
      <div
        class="sc-bczRLJ fyHqCQ"
        font-family="secondary"
      >
        We will send a text with a one time password to *** *** 
        .
      </div>
    </div>
    <div
      class="sc-bczRLJ sc-gsnTZi jIxXay hzPtvw"
      width="100%"
    >
      <button
        class="sc-bczRLJ iElVFP"
        font-size="16px"
        height="48px"
        width="100%"
      >
        SHOW
         VERIFICATION CODE
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`VerifyUser Component when user is not authorized(admin) should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ sc-gsnTZi doahKO hzPtvw"
        height="100%"
        width="100%"
      >
        <img
          class="sc-bczRLJ iQjTFr"
          src="Illustration_Phone-OTP.svg"
        />
        <div
          class="sc-bczRLJ bOThQT"
        >
          Making sure it's you
        </div>
        <div
          class="sc-bczRLJ fyHqCQ"
          font-family="secondary"
        >
          We will send a text with a one time password to *** *** 
          .
        </div>
      </div>
      <div
        class="sc-bczRLJ sc-gsnTZi jIxXay hzPtvw"
        width="100%"
      >
        <button
          class="sc-bczRLJ iElVFP"
          font-size="16px"
          height="48px"
          width="100%"
        >
          SEND
           VERIFICATION CODE
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ sc-gsnTZi doahKO hzPtvw"
      height="100%"
      width="100%"
    >
      <img
        class="sc-bczRLJ iQjTFr"
        src="Illustration_Phone-OTP.svg"
      />
      <div
        class="sc-bczRLJ bOThQT"
      >
        Making sure it's you
      </div>
      <div
        class="sc-bczRLJ fyHqCQ"
        font-family="secondary"
      >
        We will send a text with a one time password to *** *** 
        .
      </div>
    </div>
    <div
      class="sc-bczRLJ sc-gsnTZi jIxXay hzPtvw"
      width="100%"
    >
      <button
        class="sc-bczRLJ iElVFP"
        font-size="16px"
        height="48px"
        width="100%"
      >
        SEND
         VERIFICATION CODE
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
